web/src/message_view.ts

```diff
@@ -180,6 +185,10 @@ function create_and_update_message_list(
         if (!filter.requires_adjustment_for_moved_with_target) {
             const superset_datasets = message_list_data_cache.get_superset_datasets(filter);
             for (const superset_data of superset_datasets) {
+                // Reset properties that might have been set.
+                id_info.final_select_id = undefined;
+                id_info.first_unread_msg_id_pending_server_verification = -1;
```

Comment:
Eek, did we have a previous bug where `id_info.final_select_id` hadn't been being cleared? That probably deserves to be its own prep commit, since it likely fixes some other hard-to-understand bug.

Reply:
I suppose `id_info.local_select_id` would also require this, if we didn't break when that's set? That's worth mentioning in a comment here.

Reply:
Yeah, added it as a separate commit to reset the whole `id_info` here.